ext/flash/lib.rs

```diff
@@ -1139,16 +1178,57 @@ fn run_server(
             content_read: 0,
             content_length,
             expect_continue,
+            socket_rx,
+            owned_socket: None,
           })
           .ok();
+
+          socket_senders.insert(token, socket_tx);
         }
       }
     }
   }
 
+  // Now the flash thread is about to finish, but there may be some unsettled
+  // promises in the main thread that will use the socket. To make the socket
+  // alive longer enough, we move its ownership to the main thread.
+  for (tok, socket) in sockets {
+    if let Some(sender) = socket_senders.remove(&tok) {
+      // Do nothing if the receiver has already been dropped.
+      _ = sender.send(socket);
+    }
+  }
+
```

Comment:
This is key; to prevent sockets from being released, transmit them to each ServerContext. This ensures that unsettled promises that outlive the flash thread can use sockets later.